cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdminTest.java

```diff
@@ -372,7 +372,7 @@ public void testDataDirCreation() throws Exception {
     // Note: this is only meaningful if the user running this test is in at least 2 groups
     String[] groups = UserGroupInformation.getCurrentUser().getGroupNames();
     Assert.assertTrue(groups.length > 0);
-    String nsGroup = groups[groups.length - 1];
+    String nsGroup = groups[groups.length > 1 ? 1 : 0];
```

Comment:
similarly, needed to fix a unit test failure. Sometimes. the UGI contains groups that the user does not actually belong to... chosing the second group instead of the last fixed that, and does not reduce the validity of the tets

Reply:
Can you add this as a comment in the code as well.